biz.aQute.bnd.reporter/bnd.bnd

```diff
@@ -30,7 +30,8 @@
 	org.objectweb.asm;version="[5.0.3,6.0.0)",\
 	org.objectweb.asm.analysis;version="[5.0.3,6.0.0)",\
 	org.objectweb.asm.tree;version="[5.0.3,6.0.0)",\
-	org.objectweb.asm.util;version="[5.0.3,6.0.0)"
+	org.objectweb.asm.util;version="[5.0.3,6.0.0)",\
+	com.github.javaparser.javaparser-core;version=3.13
```

Comment:
I am getting concerned that the reporter is dragging in more and more 3rd party dependencies. Since all of these are included in biz.aQute.bnd.jar, it gets larger and larger. The reporter is by far the biggest adder of new 3rd party dependencies to the Bnd project.

jtwig, parboiled, apache commons, guava, asm and now javaparser.

Reply:
I have the same concerns.

Reply:
Do you think it is possible to move plugins that bring those dependencies (I think of the JtwigTransformer and now the CodeSnippet plugin) into a dedicated project such as `biz.aQute.bnd.reporter.extra` and let user add this bundle to the class path when he uses the bnd CLI (to get the extra features). For maven is not a problem no? As it is a separate plugin. Or is it an issue at the workspace level too?

Reply:
I don't see that this helps bnd as a command line tool. It is currently installable from brew as a standalone jar.

We could make the reporter jar its own command line tool (have a main), but it would then need to download and class load its own deps.

Also, the deps in the maven plugin are not without risk of conflict with the deps of other maven plugins in a user's build.

Reply:
Hmm, deps in maven can be applied in isolation like so:

```xml
<plugin>
    ...
    <dependencies>
       ...
    </dependencies>
</plugin>
```
so the risk is low to collide with other plugin dependencies

Reply:
but in gradle there is no such magic, there you could be in trouble.

Reply:
So maven gives each plugin its own class loader? Did not realize that.

Reply:
correct!

Reply:
Should I still work on this PR? I think after this one I will not add more features that need 3rd party dependencies, I could add a comment in the bnd.bnd to recall to not add extra deps and thus postpone this issue in the future.
If it is already an issue, I can:
* In bnd CLI, check if it is possible to precisely import what we need (`-conditionalpackage`) for Jtwig and JavaParser.
* In Maven, apparently it is not a problem.
* In Gradle, the plugin does not exist yet.
* Is there a benefice for maintenance and clarity to create a dedicated repository (cnf/ext/reporter.mvn)?

Reply:
Lets leave this PR chill for a little while until the Bnd maintainers can discuss at our next call.